### DISCUSSION TITLE
Sparkline examples copied directly from Textual Guide do not "render as advertised"

### DISCUSSION THREAD

**Discussion by traveller1011** [Textualize/textual#4347] [Q&A] [answered]
#### Problem
Both examples provided in the ```Sparkline section``` of the Textual guide does not render as advertised.

#### What is shown in the guide
https://textual.textualize.io/widgets/sparkline/#__tabbed_2_1

![image](https://github.com/Textualize/textual/assets/93844523/a9c46b95-f032-4728-b09c-4b1d60546a61)

#### First Docs Example
- what appears on my terminal (running from PowerShell on Windows 10 Pro)
![image](https://github.com/Textualize/textual/assets/93844523/0a172f75-7b4f-4c55-ac7e-204c9d1f595b)

- Notably if i increase the css width attribute for Sparkline from ```3``` to ```40```...

```
- Screen {
    align: center middle;
}

Sparkline {
    width: 40;    <<< this is 3 in the example
    margin: 2;
}
```
... yields this on my machine:
![image](https://github.com/Textualize/textual/assets/93844523/8ff494ab-421a-4fe0-8b00-7f03242d749d)

#### Second Docs Example (Sparkline Colors)
- Same situation... this is what shows for me:
![image](https://github.com/Textualize/textual/assets/93844523/5ba71c14-10a1-48fb-91c9-73923b442cd1)

**Comment by davep** [comment]
You appear to have an issue with your choice of terminal and/or font. Perhaps [this search of discussions will help](https://github.com/Textualize/textual/discussions?discussions_q=is%3Aopen+windows+font)?

**Reply by traveller1011** [reply]
dude if this also solves the problem of ? appearing on my plotext charts ill be super delighted. researching into the font issue and will report back one way or another. thank you.

**Comment by traveller1011** [comment, marked as answer]
Here is how to fix Sparklines and possibly other widget renderings, in Windows 10 pro.

1. install the ```DejaVuSans Mono font``` recommended in this related discussion ( https://github.com/Textualize/textual/discussions/3117 )
a) font source: https://www.nerdfonts.com/font-downloads

2. Native Powershell is not amenable to changing Fonts without Registry Edit, which possibly is fine but I did not want to do it. Also also native powershell doesnt have a fallback font in cases a character is missing.

3. For these reasons I used installed Windows Terminal which was also recommended in Textual discussions 

4. Install the Regular Mono version of the DejaVuSans font onto the system by unzipping and double clicking the corresponding font file. Had never installed a custom font before.

6. Applied both fonts in windows Terminal

![image](https://github.com/Textualize/textual/assets/93844523/fdcea642-6c6b-4a80-bc78-032ab0764bb3)

This fixed my Sparklines rendering problems 100%. Hopefully helps someone else.

**Reply by TomJGooding** [reply]
Are you sure you're running the right example here..? This looks like the first `sparkline_basic.py` but you're expecting the next one down...?

EDIT: Nevermind I see you figured it out!

**Reply by traveller1011** [reply]
yes as usual the broken computer was the one between my ears. I updated my remarks above to reflect what I did in case this is useful for others in the future. Thanks @davep and @TomJGooding . Have a great day!

**Comment by traveller1011** [comment]
-hmmm well, the second ```sparkline_colors``` actually looks pretty decent for me now.

![image](https://github.com/Textualize/textual/assets/93844523/05ed6b75-671d-4bb5-bc46-e6a869452d58)

.... compared to the textual guide:

https://textual.textualize.io/widgets/sparkline/#__tabbed_2_1

![image](https://github.com/Textualize/textual/assets/93844523/0e5b3fbc-841a-4fe0-96f9-56f8d1b9e780)

the delta between the rendering of the first sparkline example in the textual guide vs my own application is still pretty huge. will update if i find more.

**Reply by davep** [reply]
What "delta"?

**Reply by traveller1011** [reply]
omg there are **three** sparkline examples not **two**. I was "comparing" the output of example two using the source code from example one. 

EVERYTHING IS RENDERING PERFECTLY. Thanks Dave!!!

![image](https://github.com/Textualize/textual/assets/93844523/87051f10-398e-44d7-926b-629fb7a271a0)

**Reply by davep** [reply]
Glad you got it sorted. 🙂
